evm-contracts/src/v0.6-ovm/AggregatorProxy.sol

```diff
@@ -0,0 +1,444 @@
+    public
+    view
+    virtual
+    override
+    returns (uint256 updatedAt)
+  {
+    if (_roundId > MAX_ID) return 0;
+
+    (uint16 phaseId, uint64 aggregatorRoundId) = parseIds(_roundId);
+    AggregatorV2V3Interface aggregator = phaseAggregators[phaseId];
+    if (address(aggregator) == address(0)) return 0;
+
+    return aggregator.getTimestamp(aggregatorRoundId);
+  }
+
+  /**
+   * @notice get the latest completed round where the answer was updated. This
+   * ID includes the proxy's phase, to make sure round IDs increase even when
+   * switching to a newly deployed aggregator.
+   *
+   * @dev #[deprecated] Use latestRoundData instead. This does not error if no
+   * answer has been reached, it will simply return 0. Either wait to point to
+   * an already answered Aggregator or use the recommended latestRoundData
+   * instead which includes better verification information.
+   */
+  function latestRound()
+    public
+    view
+    virtual
+    override
+    returns (uint256 roundId)
+  {
+    Phase memory phase = currentPhase; // cache storage reads
+    return addPhase(phase.id, uint64(phase.aggregator.latestRound()));
+  }
+
+  /**
+   * @notice get data about a round. Consumers are encouraged to check
+   * that they're receiving fresh data by inspecting the updatedAt and
+   * answeredInRound return values.
+   * Note that different underlying implementations of AggregatorV3Interface
+   * have slightly different semantics for some of the return values. Consumers
+   * should determine what implementations they expect to receive
+   * data from and validate that they can properly handle return data from all
+   * of them.
+   * @param _roundId the requested round ID as presented through the proxy, this
+   * is made up of the aggregator's round ID with the phase ID encoded in the
+   * two highest order bytes
+   * @return roundId is the round ID from the aggregator for which the data was
+   * retrieved combined with an phase to ensure that round IDs get larger as
+   * time moves forward.
+   * @return answer is the answer for the given round
+   * @return startedAt is the timestamp when the round was started.
+   * (Only some AggregatorV3Interface implementations return meaningful values)
+   * @return updatedAt is the timestamp when the round last was updated (i.e.
+   * answer was last computed)
+   * @return answeredInRound is the round ID of the round in which the answer
+   * was computed.
+   * (Only some AggregatorV3Interface implementations return meaningful values)
+   * @dev Note that answer and updatedAt may change between queries.
+   */
+  function getRoundData(uint80 _roundId)
+    public
+    view
+    virtual
+    override
+    returns (
+      uint80 roundId,
+      int256 answer,
+      uint256 startedAt,
+      uint256 updatedAt,
+      uint80 answeredInRound
+    )
+  {
+    (uint16 phaseId, uint64 aggregatorRoundId) = parseIds(_roundId);
+
+    (
+      uint80 roundId,
```

Comment:
Why declare new variables here? Isn't this shadowing? Seems like it should cause a linting error.